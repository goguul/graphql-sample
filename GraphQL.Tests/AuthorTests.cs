using GraphQL.Authors;
using GraphQL.Books;
using GraphQL.Data;
using HotChocolate;
using HotChocolate.Data;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace GraphQL.Tests;

public class AuthorTests
{
    [Fact]
    public async Task Author_Schema_Changed()
    {
        ISchema schema = await new ServiceCollection()
            .AddPooledDbContextFactory<LibraryDbContext>(
                options => options.UseInMemoryDatabase("Data Source=library.db"))
            .AddGraphQL()
            .RegisterDbContext<LibraryDbContext>(DbContextKind.Pooled)
            .AddMutationConventions()
            .AddQueryType()
            .AddMutationType()
            .AddTypeExtension<AuthorQuery>()
            .AddTypeExtension<AuthorMutation>()
            .AddTypeExtension<AuthorType>()
            .AddTypeExtension<AuthorInputType>()
            .AddGlobalObjectIdentification()
            .AddFiltering()
            .AddSorting()
            .AddProjections()
            .BuildSchemaAsync();

        schema.Print().MatchSnapshot();
    }

    [Fact]
    public async Task Add_Author()
    {
        IRequestExecutor requestExecutor = await new ServiceCollection()
            .AddPooledDbContextFactory<LibraryDbContext>(
                options => options.UseInMemoryDatabase("Data Source=library.db"))
            .AddGraphQL()
            .RegisterDbContext<LibraryDbContext>(DbContextKind.Pooled)
            .AddMutationConventions()
            .AddQueryType()
            .AddMutationType()
            .AddTypeExtension<BookType>()
            .AddTypeExtension<AuthorQuery>()
            .AddTypeExtension<AuthorMutation>()
            .AddTypeExtension<AuthorType>()
            .AddTypeExtension<AuthorInputType>()
            .AddGlobalObjectIdentification()
            .AddFiltering()
            .AddSorting()
            .AddProjections()
            .BuildRequestExecutorAsync();

        IExecutionResult executionResult = await requestExecutor.ExecuteAsync(@"
            mutation AddAuthor {
                addAuthor(
                    input: {
                        Name: ""Test Author""
                    }) {
                        Id,
                        Name
                    }
                )
            }
        ");

        executionResult.ToJson().MatchSnapshot();
    }
}