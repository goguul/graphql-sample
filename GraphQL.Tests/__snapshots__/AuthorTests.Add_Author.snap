{
  "errors": [
    {
      "message": "Expected a `Name`-token, but found a `RightParenthesis`-token.",
      "locations": [
        {
          "line": 10,
          "column": 17
        }
      ],
      "extensions": {
        "code": "HC0014"
      }
    }
  ]
}
